src/components/Checkbox/index.tsx

```diff
@@ -60,9 +60,10 @@ const Checkbox = ({ className, ...otherProps }: InternalCheckboxProps) => {
 };
 
 export interface InternalCheckboxProps
-	extends DefaultProps,
+	extends Omit<DefaultProps, 'defaultValue'>,
```

Comment:
Sorry about this, but what I think should have happened, is in `TableRow`, `checkboxAttributes` should be of `CheckboxProps` type instead, which will resolve this issue and make it more accurate.

Reply:
Hm, still doesn't work. Wondering how did the last PR get through.

Reply:
@dimitrisnl You also need to update `Table.index`, as the `checkboxAttributes` comes from there.